AppCenterPush/AppCenterPushTests/MSPushTests.m

```diff
@@ -119,6 +119,36 @@ - (void)testApplyEnabledStateWorks {
   XCTAssertTrue([service isEnabled]);
 }
 
+- (void)testApplyEnabledAddsAndRemovesDelegate {
```

Comment:
I would divide this test into two tests: enabling adds delegate, disabling removes delegate. This will simplify things and will cover more ground for instance in your test we don't know when the delegate was added and removed.